hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -865,29 +861,39 @@ public boolean seekBefore(Cell key) throws IOException {
         // The key we are interested in
         if (previousBlockOffset == -1) {
           // we have a 'problem', the key we want is the first of the file.
+          releaseIfNotCurBlock(seekToBlock);
           return false;
         }
 
         // The first key in the current block 'seekToBlock' is greater than the given
         // seekBefore key. We will go ahead by reading the next block that satisfies the
         // given key. Return the current block before reading the next one.
-        seekToBlock.release();
```

Comment:
This fix is there as part of some other issue fix right?

Reply:
Yeah, similar to the issue: https://issues.apache.org/jira/browse/HBASE-22480.  Discussed with binlijin before, his patch can apply to branch-2.x and need not to apply to master branch,  once the HBASE-21879 get merged into master, then this bug will be fixed in this patch.